### PR TITLE
feat: implement `talosctl config new` command

### PR DESCRIPTION
**Pull request by AlekSi**
Refs #3421.

~I wonder if `talosctl talosconfig` is the best name and location for that command. Could it be `talosctl gen talosconfig`? `talosctl config gen`? `talosctl config new`?~

It is `talosctl config new` now.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/3761)

### CODE REVIEW

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go [hunk 1]
```diff
@@ -1916,6 +1918,47 @@ func (s *Server) RemoveBootkubeInitializedKey(ctx context.Context, in *empty.Emp
...
+	input.Certs.Admin = cert
+	talosconfig, err := generate.Talosconfig(input)
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := talosconfig.Bytes()
+	if err != nil {
+		return nil, err
+	}
```
Comment (AlekSi) [inline internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:1946]: Hot mess :)

File: .dockerignore [hunk 2]
```diff
@@ -5,7 +5,6 @@
 !hack
 !internal
 !pkg
-!vendor
```
Comment (AlekSi) [inline .dockerignore:8]: Useful for quick local hacks

File: api/cluster/cluster.proto [hunk 3]
```diff
@@ -3,9 +3,6 @@ syntax = "proto3";
 package cluster;
 
 option go_package = "github.com/talos-systems/talos/pkg/machinery/api/cluster";
-option java_multiple_files = true;
-option java_outer_classname = "ClusterApi";
-option java_package = "com.cluster.api";
```
Comment (AlekSi) [inline api/cluster/cluster.proto:8]: Those options were not present everywhere (health.proto, for example), and cluster.com costs $150k at least: https://domaininvesting.com/cluster-com-acquired-by-namecheap-for-150k/

File: internal/app/apid/main.go [hunk 4]
```diff
@@ -46,14 +54,7 @@ func Main() {
...
 
-	go func() {
-		debugLogFunc := func(msg string) {
-			log.Print(msg)
-		}
-		if lErr := debug.ListenAndServe(context.TODO(), debugAddr, debugLogFunc); lErr != nil {
-			log.Fatalf("failed to start debug server: %s", lErr)
-		}
-	}()
+	go runDebugServer(context.TODO())
```
Comment (AlekSi) [inline internal/app/apid/main.go:57]: Extracted to reduce cyclomatic complexity without slapping `//nolint:gocyclo` on it

File: internal/app/apid/pkg/backend/apid.go [hunk 5]
```diff
@@ -54,7 +54,7 @@ func (a *APID) String() string {
 // GetConnection returns a grpc connection to the backend.
 func (a *APID) GetConnection(ctx context.Context) (context.Context, *grpc.ClientConn, error) {
 	md, _ := metadata.FromIncomingContext(ctx)
-	md = metadata.Join(md, authz.RolesAsMetadata(authz.GetRoles(ctx)))
+	md = metadata.Join(md, authz.RolesAsMetadata(authz.GetRoles(ctx))) // TODO(rbac): duplicates?
```
Comment (AlekSi) [inline internal/app/apid/pkg/backend/apid.go:57]: It seems that metadata is being duplicated there or in Local's method with the same name. It does not cause problems but means that my mental model of how everything works is wrong. I will fix it and add tests in the next PR.

File: pkg/machinery/config/types/v1alpha1/generate/generate.go [hunk 6]
```diff
@@ -383,13 +383,10 @@ func NewTalosCA(currentTime time.Time) (ca *x509.CertificateAuthority, err error
 }
 
 // NewAdminCertificateAndKey generates the admin Talos certificate and key.
-func NewAdminCertificateAndKey(currentTime time.Time, ca *x509.PEMEncodedCertificateAndKey, loopback string, role role.Role) (p *x509.PEMEncodedCertificateAndKey, err error) {
-	ips := []net.IP{net.ParseIP(loopback)}
-
+func NewAdminCertificateAndKey(currentTime time.Time, ca *x509.PEMEncodedCertificateAndKey, roles role.Set, ttl time.Duration) (p *x509.PEMEncodedCertificateAndKey, err error) {
 	opts := []x509.Option{
-		x509.Organization(string(role)),
-		x509.IPAddresses(ips),
```
Comment (AlekSi) [inline pkg/machinery/config/types/v1alpha1/generate/generate.go:391]: IPs do not make sense there (?)

File: cmd/talosctl/cmd/mgmt/gen/csr.go [hunk 7]
```diff
@@ -55,9 +55,9 @@ var genCSRCmd = &cobra.Command{
 			return fmt.Errorf("invalid IP: %s", genCSRCmdFlags.ip)
 		}
 
-		roles, err := role.Parse(genCSRCmdFlags.roles)
+		roles, unknownRoles := role.Parse(genCSRCmdFlags.roles)
 		if err != nil {
```
Comment (smira) [inline cmd/talosctl/cmd/mgmt/gen/csr.go:59]:
```suggestion
		if len(unknownRoles) > 0 {
```

**Review by smira — APPROVED** [review]
one nit/bug :+1:

**Review by smira — APPROVED** [review]

**Pull request merged**